engine/include/gen/application.hpp

```diff
@@ -4,27 +4,29 @@
 
 #include "core.hpp"
 
-#include "util/version.hpp"
+#include "gen/util/version.hpp"
 
-#include <logger/log.hpp>
-#include "graphics/device.hpp"
-#include "windowing/window.hpp"
+#include "gen/logger/log.hpp"
+#include "gen/windowing/window.hpp"
+
+#include "engine.hpp"
+#include "time.hpp"
 
 namespace gen
 {
 	class Application
 	{
 	public:
 		explicit Application(const char * appName, mim::vec2i const & initialSize);
+		virtual ~Application();
+
+		virtual void run() final;
 
-		void run();
+		virtual void draw();
```

Comment:
Generally `draw` means one draw call, or at most one set of primitives; better to use `render` here as the entire scene will be rendered rather than just a specific render object.